snorkel/matchers.py

```diff
@@ -225,8 +242,8 @@ def init(self):
 
         # Compile regex matcher
         # NOTE: Enforce full span matching by ensuring that regex ends with $!
-        self.rgx = self.rgx if self.rgx.endswith('$') else self.rgx + r'$'
-        self.r = re.compile(self.rgx, flags=re.I if self.ignore_case else 0)
+        self.rgx = self.rgx if self.rgx.endswith('$') else ('(?:' + self.rgx + ')' + '$')
```

Comment:
Added a comment. We group self.rgx first so that the $ applies to all components of an 'OR' expression, and not just the last one. (e.g., we want r'(a|b)$' instead of r'a|b$')

Reply:
Is there a reason we use the non-capture group `?:`?

Reply:
No reason. I think it doesn't make a difference either way here, since we take the full match. We could drop the '?:' for simplicity. I'll go ahead and do that.